Added documentation for new integration Loqed (#22439)
home-assistant/home-assistant.github.io: source/_integrations/loqed.markdown

---
title: Loqed Touch Smart Lock
description: Instructions on how to integrate a Loqed Touch Smart Lock
ha_category:
  - Lock
ha_release: 2023.7
ha_iot_class: Local Push
ha_codeowners:
  - "@mikewoudenbergs"
ha_domain: loqed
ha_platforms:
  - lock
  - sensor
ha_config_flow: true
ha_dhcp: true
ha_integration_type: integration
---

Integrate your LOQED Touch Smart Lock with Home Assistant. The lock instantly notifies Home Assistant of a lock state change and you can change the lock state yourself.

## Features

This integration supports:

- Send real-time status changes of the lock (open, unlock, lock)
- Change the lock state (open, unlock, lock).
  - Only if your lock has a fixed knob on the outside of your door, you can use the “open” lock state. If you do not have this (thus you have a handle on the outside of your door that you can push down), this command will behave as if the unlock command is sent.

## Prerequisites

On the [LOQED personal access token website](https://integrations.production.loqed.com/personal-access-tokens), please follow the following steps:

{% details "Generate access token" %}

> 1. Login with your LOQED App e-mail address (you need to be admin)
> 2. Tap on "Create"
> 3. Give your personal access token a name (this will not be used further on, but we recommend something like "Home Assistant" to be able to recognize it as used by Home Assistant)
> 4. Tap on Save
> 5. Store the access token somewhere you can easily copy/paste from as you'll need them in the next step (and it will only be shown once). Note that you can use this token for setting up multiple locks.
>    {% enddetails %}

Reviewer: ```suggestion
1. Login with your LOQED App e-mail address (you need to be an admin).
2. Select **Create**.
3. Give your personal access token a name (this will not be used further on, but we recommend something like "Home Assistant" to recognize it as used by Home Assistant).
4. Select **Save**.
5. Store the access token somewhere you can easily copy/paste from, as you'll need them in the next steps (and it will only be shown once). Note: that you can use this token for setting up multiple locks.
   {% enddetails %}
```

{% include integrations/config_flow.md %}

> Home Assistant should automatically detect your lock when your Home Assistant runs on the same network as your lock. In that case you only need to provide the selected API key when you configure the integration.
> 
> You can also set up a lock manually when for some reason it is not automatically detected. In that case you need to provide both the API Key from the previous step and the name of the Lock as it is known in the LOQED companion app.

Reviewer: ```suggestion
Home Assistant should automatically detect your lock when your Home Assistant runs on the same network as your lock. In that case, you only need to provide the selected API key when configuring the integration.

You can also set up a lock manually when for some reason, it is not automatically detected. In that case, you need to provide both the API Key from the previous step and the name of the Lock, as it is known in the LOQED companion app.
```

## Services

Please see the default [lock integration page](/integrations/lock/) for the services available for the lock.

## De-installation in Loqed

> First remove the integration from Home Assistant. This will take care of removing any configuration made on the lock itself for Home Assistant.

Reviewer: ```suggestion
First, remove the integration from Home Assistant. This will remove any configuration made on the lock itself for Home Assistant.
```

On [LOQED personal access token website](https://integrations.production.loqed.com/personal-access-tokens), please follow the following steps:

> - Login with your LOQED App e-mail address (you need to be admin)
> - Tap delete on the Personal Access Token you used when creating this integration.

Reviewer: ```suggestion
1. Login with your LOQED App e-mail address (you need to be admin).
2. Select **delete** on the Personal Access Token you used when creating this integration.
```

---
title: Loqed Touch Smart Lock
description: Instructions on how to integrate a Loqed Touch Smart Lock
ha_category:
  - Lock
ha_release: 2022.7
ha_iot_class: Local Push
ha_codeowners:
  - "@cpolhout"
ha_domain: loqed
ha_platforms:
  - lock
  - sensor
ha_config_flow: true
ha_dhcp: true
ha_integration_type: integration
---

Integrate your LOQED Touch Smart Lock with Home Assistant. The lock instantly notifies Home Assistant of a lock state change (no polling), and you can change the lock state yourself.

## Features

This integration supports:

- Send real-time (!) status changes of the lock (open, unlock, lock)
- Send battery status updates
- Change the lock state (open, unlock, lock).

>   - IMPORTANT: The Home Assistant Lovelace standard card has only support for unlock and lock. By leveraging the lock-service: lock.open you can connect any button or action to opening the lock.

Reviewer: This is not an integration concern.

```suggestion
```

Reviewer 2: Removed

  - Only if your lock has a fixed knob on the outside of your door, you can use the “open” lock state. If you do not have this (thus you have a handle on the outside of your door that you can push down), this command will behave as if the unlock command is sent.

> - Future: change certain settings on the LOQED Touch Smart Lock

Reviewer: Future? As in? You can't right now? If that is the case, let's remove it.

Reviewer 2: It's indeed future functionality. Removed it from the docs

- Switch the relay in the bridge (to open a shared access door or garage door)

## Prerequisites

> On https://app.loqed.com/API-Config/, please follow the following steps:

Reviewer: Please don't use bare URLs

Reviewer 2: Fixed

- Login with your LOQED App e-mail address (you need to be admin)
- Tap “API Configuration tool”
- Tap “Add new API key”
- Choose any name and your lock, and tap “Add API key”. If you cannot add an API key, ensure that your lock is connected to the internet (the LED on your bridge should be constant green).
  Back on the overview page, under the heading “API Keys”, tap the button “View / Edit” next to your newly created API key.
- Copy the contents of the field “Integration information”. This starts with “{"lock_id":"....”

> NB 1: You do not need to create any webhooks (not for the web API, nor for the bridge), as the LOQED integration will take care of this.

Reviewer: I'm confused, what does this even mean?

Reviewer 2: Originally the user needed to create webhooks in the LOQED webapp manually to integrate the lock into home assistant. This new integration takes care of that automatically. Removed the section

> NB 2: The API call from the LOQED lock to Home Assistant is verified before the lock-status is updated and the event is generated. This is to prevent incoming calls to change the lock-status.

Reviewer: For what is this relevant? I think we can just remove this part.

Reviewer 2: Removed

{% include integrations/config_flow.md %}

## Services

> Please see the default lock integration for the services: https://www.home-assistant.io/integrations/lock/

Reviewer: Please don't use bare URLs, and don't use full links to link to other parts of the documentation (instead use links with paths from the root domain).

Reviewer 2: Adjusted the link and added a descriptive text

## De-installation in Loqed

> On https://app.loqed.com/API-Config/, please follow the following steps:

Reviewer: Please don't use bare URLs

Reviewer 2: Adjusted

- Login with your LOQED App e-mail address (you need to be admin)
- Tap “API Configuration tool”
- Under the heading “API Keys”, remove the API key you created previously.
- Ensure your computer is connected to your local home network. Under the heading “Outgoing Webhooks via LOQED Bridge”, tap “Add/delete webhooks” next to your LOQED Bridge. On the next page, ensure to remove all webhooks from your LOQED Bridge.

# Security

> All commands that are sent to the lock contain a digital signature to prevent replay attacks. Also the webhooks that the LOQED Bridge sends to notify Home Assistant of a lock state change are digitally signed. As there is no TLS encryption to the bridge, people with access to your local network could potentially see the communication, but it cannot be altered.

Reviewer: ```suggestion
All commands that are sent to the lock contain a digital signature to prevent replay attacks. Also, the webhooks that the LOQED Bridge sends to notify Home Assistant of a lock state change are digitally signed. As there is no TLS encryption to the bridge, people with access to your local network could potentially see the communication, but it cannot be altered.
```

Reviewer 2: Removed altogether as it indeed is a technical detail irrelevant to the cosumer of the integration

> Via https://app.loqed.com/API-Config/ you have copy the config containing two different keys during setup: one key which is used to sign commands between you and the bridge (this key only changes if you re-install your lock), and one key to sign commands you send to the lock. The latter key is not accessible by LOQED (it’s encrypted with your LOQED account password, of which only a hash is stored on LOQED’s server). The LOQED Home Assistant integration uses this key to sign commands to the lock (e.g. to unlock). Thus, LOQED’s security architecture, where the keys are only stored on your phone and the lock (and now also on your Home Assistant system) is preserved.

Reviewer: - Don't use bare URLs
- `copy` -> `copied`

Honestly, this whole section sounds like jibberish to most users, and I'm not sure what it adds to the integration documentation to have such a level of detail.

Reviewer 2: Removed altogether as it indeed is a technical detail irrelevant to the cosumer of the integration
